mx.py

```diff
@@ -3442,20 +3442,23 @@ def pull(self, vcdir, rev=None, update=False, abortOnError=True):
         :return: True if the operation is successful, False otherwise
         :rtype: bool
         """
-        rc = self._fetch(vcdir, abortOnError)
-        if rc == 0 and update:
+        if update and not rev:
             active_branch = self._active_branch(vcdir, abortOnError)
-            refspec = '{0}:{1}'.format(rev if rev else 'HEAD', active_branch)
-            cmd = ['git', 'pull', 'origin', refspec]
+            cmd = ['git', 'pull', 'origin', 'HEAD:{0}'.format(active_branch)]
             self._log_pull(vcdir, rev)
             out = OutputCapture()
             rc = self.run(cmd, nonZeroIsFatal=abortOnError, cwd=vcdir, out=out)
             logvv(out.data)
             return rc == 0
         else:
-            if abortOnError:
-                abort('fetch returned ' + str(rc))
-            return False
+            rc = self._fetch(vcdir, abortOnError)
+            if rc == 0:
+                if rev:
+                    return self.update(vcdir, rev=rev, mayPull=False, clean=True, abortOnError=abortOnError)
```

Comment:
I'm not sure about the `clean=True` that looks dangerous.
In the `hg` implementation we would not clean and rather abort if we can not update cleanly.

Also we should only update if `update` is true